### PR TITLE
Carbon Styles: Sass style sheet

### LINKED ISSUE #633: Carbon: Investigate an alternative for `styled-components`
### What should we do?

For the `form-js-carbon-styles`, we started to use `styled-components`. This speeds up development as we simply mirrored the styles from Tasklist, the only use case we currently have for these styles (cf. https://github.com/camunda/team-hto/issues/143).

We should invest in an alternative to not tight these style to React applications. 

### Why should we do it?

Follow up of #557 

We should not rely on such a specific library, but cf. to a more basic approach, e.g. CSS or SCSS.

### PR DESCRIPTION
**Pull request by pinussilvestrus**
Closes #633 

* Transforms the styled component `carbon-styles` to a `scss` file
* Adds a visual regression test that uses the `scss` file
* Deprecate old `carbon-styles.js` styled component

It can be tested via `npm run start:carbon`

### CODE REVIEW

**Comment by pinussilvestrus** [comment]
I'd like to update the `theming` specs to use the carbon styles as well. It anyway already uses the `/carbon` root.

**Comment by pinussilvestrus** [comment]
@vsgoulart the transformation should be done 🎉  

Things to clarify
* I for now only deprecated the old styled component, and removed the reference from the package. We could also remove it right now, but we can also wait to support the legacy use case at least.
* The Tasklist Carbonization is failing because of the point above. I will try out the new sass file in Tasklist and see whether everything works as expected 👍

**Comment by pinussilvestrus** [comment]
Some minor things are broken with the sass file

image

I will handle these 👍 

Here is the draft PR in Tasklist to replace the stylesheet: https://github.com/camunda/tasklist/pull/3749

**Comment by pinussilvestrus** [comment]
The minor issues are resolved ✅

**Review by vsgoulart — APPROVED** [review]
Tested it on Tasklist and everything worked fine

File: packages/form-js/package.json [hunk 1]
```diff
@@ -24,7 +24,7 @@
     "./dist/assets/form-js-editor.css": "./dist/assets/form-js-editor.css",
     "./dist/assets/form-js-editor-base.css": "./dist/assets/form-js-editor-base.css",
     "./dist/assets/form-js-playground.css": "./dist/assets/form-js-playground.css",
-    "./dist/carbon-styles": "./dist/carbon-styles.js",
+    "./dist/assets/carbon-styles.scss": "./dist/assets/carbon-styles.scss",
```
[the remark this answers is not in the document]
Comment (pinussilvestrus) [reply packages/form-js/package.json:27]: Yeah I thought about that one, but as you said, the main use case currently is Tasklist and our products anyway, so a CSS file would be obsolete. We can wait for feedback and easily add the compilation step on top.

**Pull request merged**